infra: bump introspector

DavidKorczynski opened the pull request:
This has a set of performance improvements in Fuzz Introspector, the two changes with most impact are:
- removal of some expensive and unnecessary loops in the code
- switching parsing of large yaml files from pure python code to using a C backend.

Locally it makes OpenSSL builds take approximately 70 minutes whereas in the cloud build it seems to take 20+ hours. Similar impact happens across several large java projects.

oliverchang commented:
/gcbrun trial_build.py all --fuzzing-engines libfuzzer --sanitizers introspector

oliverchang commented:
/gcbrun trial_build.py all --fuzzing-engines libfuzzer --sanitizers introspector

oliverchang commented:
/gcbrun trial_build.py all --fuzzing-engines libfuzzer --sanitizers introspector

jonathanmetzman commented:
David do we care about these failures?

DavidKorczynski commented:
I would say this is good to merge.

The following projects failed:
`cryptography`: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-b997a5af-4547-43c7-a4da-850c8bb59b74.txt

This is a normal build error. Not related to this PR. It was also fixed in https://github.com/google/oss-fuzz/pull/10890



`libxaac`: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-4685ffc8-1905-4595-9cf2-24849f20aece.txt

Not related to this PR.

Fails with:
```
Step #12: Digest: sha256:30b92163254edef9648158f7886b9528a9da6fa84ad7fdee37c3c24a9cbfae00
Step #12: Status: Downloaded newer image for gcr.io/oss-fuzz-base/base-builder-go-testing-bump-introspector:latest
Step #12: gcr.io/oss-fuzz-base/base-builder-go-testing-bump-introspector:latest
Finished Step #12
Step #1: CommandException: No URLs matched: gs://oss-fuzz-coverage/libxaac/textcov_reports/20230828/*
Step #1: CommandException: 1 file/object could not be transferred.
Step #6: 15e054ccd2fc: Pull complete
Finished Step #1
```

`lua`: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-66048a45-e58d-4cc5-b200-d3c1ad15717e.txt

Not related to this PR. Fails with an existing build error.

Was also fixed in: https://github.com/google/oss-fuzz/pull/10891

`lark-parser`: https://oss-fuzz-gcb-logs.storage.googleapis.com/log-a892c897-9cd3-4af4-98a7-e98381b54945.txt

Fails with a timeout. Not due to this PR.

`binutils`: 

Fails with:
```
Step #22 - "compile-libfuzzer-introspector-x86_64": Traceback (most recent call last):
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/fuzz-introspector/src/main.py", line 154, in <module>
Step #22 - "compile-libfuzzer-introspector-x86_64":     main()
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/fuzz-introspector/src/main.py", line 138, in main
Step #22 - "compile-libfuzzer-introspector-x86_64":     return_code = commands.run_analysis_on_dir(
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/fuzz-introspector/src/fuzz_introspector/commands.py", line 61, in run_analysis_on_dir
Step #22 - "compile-libfuzzer-introspector-x86_64":     introspection_proj.load_data_files(parallelise, correlation_file)
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/fuzz-introspector/src/fuzz_introspector/analysis.py", line 95, in load_data_files
Step #22 - "compile-libfuzzer-introspector-x86_64":     for idx in return_dict:
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/usr/local/lib/python3.8/multiprocessing/managers.py", line 1052, in __next__
Step #22 - "compile-libfuzzer-introspector-x86_64":     return self._callmethod('__next__', args)
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/usr/local/lib/python3.8/multiprocessing/managers.py", line 835, in _callmethod
Step #22 - "compile-libfuzzer-introspector-x86_64":     kind, result = conn.recv()
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/usr/local/lib/python3.8/multiprocessing/connection.py", line 250, in recv
Step #22 - "compile-libfuzzer-introspector-x86_64":     buf = self._recv_bytes()
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/usr/local/lib/python3.8/multiprocessing/connection.py", line 414, in _recv_bytes
Step #22 - "compile-libfuzzer-introspector-x86_64":     buf = self._recv(4)
Step #22 - "compile-libfuzzer-introspector-x86_64":   File "/usr/local/lib/python3.8/multiprocessing/connection.py", line 379, in _recv
Step #22 - "compile-libfuzzer-introspector-x86_64":     chunk = read(handle, remaining)
Step #22 - "compile-libfuzzer-introspector-x86_64": ConnectionResetError: [Errno 104] Connection reset by peer
```
which I don't think is related to this PR and seems to me a spurious issue.
